dts/riscv/sophgo/cv180x.dtsi

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023-2024 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <freq.h>
+
+#include <sophgo/cv18xx-c906-common.dtsi>
+
+/ {
+	soc {
+		compatible = "sophgo,cv180x";
+
+		mbox: mailbox@1900000 {
+			compatible = "sophgo,cvi-mailbox";
+			reg = <0x1900000 DT_SIZE_K(4)>;
+			interrupt-parent = <&plic0>;
+			interrupts = <61 1>;
+			channel-max = <8>;
+			tx-cpu = <1>;
+			rx-cpu = <2>;
+			#mbox-cells = <1>;
+			status = "disabled";
+		};
+
+		pinctrl: pin-controller@3001000 {
+			compatible = "sophgo,cvi-pinctrl";
+			reg = <0x3001000 DT_SIZE_K(4)>;
+		};
+
+		gpioa: gpio@3020000 {
+			compatible = "snps,designware-gpio";
+			reg = <0x3020000 DT_SIZE_K(4)>;
+			interrupt-parent = <&plic0>;
+			interrupts = <41 1>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			status = "disabled";
+		};
+
+		gpiob: gpio@3021000 {
+			compatible = "snps,designware-gpio";
+			reg = <0x3021000 DT_SIZE_K(4)>;
+			interrupt-parent = <&plic0>;
+			interrupts = <42 1>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			status = "disabled";
+		};
+
+		gpioc: gpio@3022000 {
+			compatible = "snps,designware-gpio";
+			reg = <0x3022000 DT_SIZE_K(4)>;
+			interrupt-parent = <&plic0>;
+			interrupts = <43 1>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			status = "disabled";
+		};
+
+		gpiod: gpio@3023000 {
+			compatible = "snps,designware-gpio";
+			reg = <0x3023000 DT_SIZE_K(4)>;
+			interrupt-parent = <&plic0>;
+			interrupts = <44 1>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			status = "disabled";
+		};
+
+		pwm0: pwm@3060000 {
+			compatible = "sophgo,cvi-pwm";
+			reg = <0x3060000 DT_SIZE_K(4)>;
+			clock-frequency = <DT_FREQ_M(100)>;
+			#pwm-cells = <3>;
+			status = "disabled";
+		};
+
+		pwm1: pwm@3061000 {
```

Comment:
please, add relevant nodes gradually as you add drivers/support/compatibles. this is throwing the entire DT in one go, which makes it difficult to review.